drizzlepac/generate_final_product_filenames.py

```diff
@@ -1,9 +1,188 @@
 #!/usr/bin/env python
 
+from astropy.table import Table, Column
+
+#Define information/formatted strings to be included in output dict
+sep_str = 'single exposure product {:02d}'
+fp_str = 'filter product {:02d}'
+tdp_str = 'total detection product {:02d}'
+
 """This script is development space for code that will be most likely
     absorbed into a larger script later on.
 
 """
+def interpret_input(results, exposure_limit=2.0):
+    """
+    Interpret the database query for a given visit to prepare the returned
+    values for use in generating the names of all the expected output products.
+    
+    Input will have format of:
+        ia1s70jtq_flt.fits,11150,A1S,70,70,149.232269,F110W,IR,/ifs/archive/ops/hst/public/ia1s/ia1s70jtq/ia1s70jtq_flt.fits
+        ia1s70iwq_flt.fits,11150,A1S,70,70,0.91161000000000003,F160W,IR,/ifs/archive/ops/hst/public/ia1s/ia1s70iwq/ia1s70iwq_flt.fits
```

Comment:
I have to ask sphinx expert @sosey to give her feedback on how to best represent these dictionaries in the doctstring. Maybe use `.. code-block::`?

Reply:
Yes, code block directive would be your best bet.

Also, if this instruction is meant for user, not developer, it is better to put this in the user-facing RST doc, not buried inside API doc.

Reply:
huh, I never got notified for this...but I agree with pllim

Reply:
The developer should always have some idea of what kind of input this code is expecting, especially when that input is NOT simple.  This allows the developer to understand what the code is trying to do, or how changes may interact with the input to avoid causing problems due to assumptions about the inputs.  Therefore, this would be a good candidate for documentation that both the developer and end user may need, so it should go in both places with a more complete explanation in the user-facing RST docs.  The use of code-block is a great suggestion in this case as well!

Reply:
This is true, however I think the code, documentation, and future users and developers would benefit if all the classes, methods, and functions in this package were updated to include the common code documentation format. Right now it appears inconsistent and there are mostly multi-line docstrings, consider updating everything to included explicit sections for attributes, outputs, notes, examples, etc., that are reST compatible. These can be automatically compiled with sphinx as part of the general documentation that is useful for everyone.